pkg/export/service.go

```diff
@@ -162,11 +162,18 @@ func (e *exporter) mqttConnect(conf config.Config, logger logger.Logger) (mqtt.C
 	opts := mqtt.NewClientOptions().
 		AddBroker(conf.MQTT.Host).
 		SetClientID(e.id).
-		SetCleanSession(true).
+		SetCleanSession(conf.MQTT.CleanSession).
 		SetAutoReconnect(true).
 		SetOnConnectHandler(e.conn).
 		SetConnectionLostHandler(e.lost)
 
+	if conf.MQTT.Persist {
+		store := mqtt.NewFileStore(conf.MQTT.PersistDir)
+		opts.SetStore(store)
+		//disable clean session because paho deletes stored messages when restarts
+		opts.SetCleanSession(false)
```

Comment:
Don't we want to make this configurable?

Reply:
if user set persistent = true but clean session = false, when service restarts messages are lost.
so i make it configurable (clean_session) in the config, but if user sets persist = true, the clean session is set accordingly.

I can change implementation and make the settings completely independent, but in that case user have to manually set clean_session = false when he wants to persist messages in fs